### PR TITLE
Adding sticky headings

### PR DESCRIPTION
**Pull request by MaathavanJkr**
Adding sticky headings

### CODE REVIEW

**Comment by jonorthwash** [comment]
Submitted to address #22.

**Comment by sushain97** [comment]
I don't think this works as expected:

image

Please also modify your JS to match the existing style.

**Comment by MaathavanJkr** [comment]
See now, i just made some changes with shuain's feedback on the issue!

**Comment by wei2912** [comment]
Saw that the indentation changed quite a lot, was this supposed to be the case? You should try to stick to the old code style, unless it's in deep need of refactoring/reformatting.

Otherwise, testing on the page uploaded to GCI seems to show that the sidebar does its intended purpose well, but there's a lot of unutilised space on the right side as a result of the adopted one-column layout.

Also, just a minor thing - could you reduce the width of the sidebar? It seems kinda large.

**Comment by sushain97** [comment]
Please post a screenshot of your changes when you'd like us to review, that'll help get feedback a bit quicker. Ditto to @wei2912 's comment about formatting. This diff has a bunch of unrelated things in it. In fact, please don't edit the old code style at all. You can replicate it with Prettier e.g. https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEB5ASgcQEEA5ASQC1CAVU7YzAXkwHIBDABzgCcYBLAVwC2zADQAdKJkwBhQtIASAUQD6AaUUBNRpgAGAEmB4iZSjToBfALRhWYABZwd4yZkUANAAqlcGtZu36hgQkFNS0xFZwAB7svFwAnk4SUu5ePsoAsqQAMtmkAMraAGwADGUlzlJU2F7ShUwA2swANqxQAOb8rO1waKIsMFz8UADW-cwY3bwd41D8XGjc8ePTYPwARqwwEFzj2xDNfQC6AHSCHAAU27FgjAB8mBfAyVKY17xgSLoc3HxClgZ3mBzEkXFIoKxBHAvkDKq8uHB2BAAKpcZpfHR2GAwdhoJAAenx7V4MDsGxOkEE+IMRhCpnC5nxPx4AkEAOAQJBcPMAEoeQBuCQSVhoeJQW4AM2GYD40EwPRguEREDQACF4lQIDcLjt2m1eAAvLa8aA8zDPFyQKAYTAOVgAE24aG0UDgAHdMPI4A6nU9MIQwGBETAvmx2Oxmh9jdB8QA3KD2k7E0nkqFcMDxSzsBGx3jugDUACs0NBmJheYLLegsDZ7HB8jBWM04NoAIQXWQKFTqLTTTDNCA2ZoNnbdOBmgA+E-7g6bI64Y4aqW8vh7R0wAB5MAARLZwE5QCBui4CoUuKXi2WSZQSuAwexKpFoC7sMdmi2vTBWm3zZoBLE4nihIcLwSYkmS6wUhAVK6mg1JBMYoRmBE+IIk+AD8r49AwBhYXA5gAGScFwyh4QwACMZQ6JWn4IjA8ySLe952Bcv4iOatreo6CzlqeLjmGeUjNlgaEquqmo3NowDmDRmAXjKJrXg69qPiqFyiWg74vFIGknBKOyKLYLGifc5rafCyonECaB6QZRlXFqHymR+n6vN+InKhJHzOkwGniY5YANEC65TpgDRHLJrmYKJXlgDZ7D8GgxnKnxUU6cqaoagFQUBeuvmeQFaCRa8FbmaV-GCZgvASo8tYOA2TbjmZYJfugBz7m6rBcFAFzMKpaAkjs8T9rwNp1S2kzNux-DsPaxodCczB8eZ7nRb0SLWi2TCsJ1JKYDed4PhlFzkalrzKMp-UXDtrB7QiaAbYsJzFtAJ5nVIq2RqMXo+lw2j3Y9+52txNkKr12TTGM72tdaWBfSMPkjd9XHcJg6FIyMP3cScD2RjAvUiEtZyXFEplRDjEYkr1-IsHymBfOFxWfSKMAeGO2jwzZzYdKSaOYK+CxwKQsAXJzenTPaFwXA0yjsQizRHGaDAPPLjAMEwzDywwYggK0GA60tDQlKc5zMRc+JkRcYhiPa+Y8viPINORiv05g5GVit1Zrec0zTO0bM9IjDQnCHhBcAu8SiyzActvmbs8icIxwPEz48qcaCRkGFwAEzQ6tAPoL02g3XtHhcNBo37k2zTqXAPtQH7Mc2ec7AXAdzH9XyTNexpiMl1gZcV091e1w9hfN5cBebaZU9PS9PV8tDve2Vwhn2G3l0ZctLUDkO86Lp2Sh+BoeWYAAUvkdA44MfvVZHflZTc0O73O2wLj0S6eCux+n7u8AHkeE8mA47LnSFkXIBRZLmEwHAQ4LYXLpSfP5SSTAL5XwFosUWs5hxvwPnII+a4zoCSgC8OiDFMCD0EJXE490DixjgLXJBj8Ph8WIcKUU4o5LSivJgJKR5Yqp2alIPQvUADE1l2KiIHA6P2RNtjtHaM2aQetnzMHtKNCMrBMyHldEtD2VZYZrSYbFYuu0sAKn6sgj4FxaQmDCHQM6q1DxcHOJGA0cB7T5G9OmOwVBohYCYCI5gojFhdXsEtcyJxYxNhPJE7Y2QjzcGkCKBh29hG9SsgFImCJBAQHoSefRVQaikDqCvNeLE-SwkwBCKEctlSoj-OYdivAlYPAQTDG0okIY2kCb1Dc-A-xgBUdrEAGkRoYEsPdO8mBBAwBKDrO4ejKrpOCdZImHBOAJiti1FZG51Gxi-MMnWQI5LNiiJYVg-BtgLPWeGBAktzKfiCXs3gsZFkJw2fc7ZaVXjPLsNnQ5Io0AjOmF9OAlh1i7xGDM+ZIB3lWX8RcGp444Q-OeawQFaBgU60EM0cibx-GWEdJABcV5LA6LgDrN4XUFQjOUJCtoIwqVax1oeLUCBuALMALwbgB0XY3EyRZ7F2k-LsAiCUXxRINNRWlXk0rPw8jlb8vporMXYpAIILgJQZnNC1ZAZolhKI3IVY8iySJukwEVcalqaS3I9wyv9DKVjArBUwKFRm5kNImpxjsfG112LrFaZgVgB5IT7hfko6CAsGGQWRXyL1EpeDNHgFwb5aU-TIvYo6NAYAuC8HYDw3k9wTWfmRZkhJbokkpJPCcVYzR+BZqRTsVxhoPFeLCb4-xk4JzFteBcLNOa808IIgRTA-bc35sUmWxJXBkmYITrW+tvRG0uKbC2zx3j7B+KiDAPklr412XXum0NmbegDondAdiWJcXKF-LxItOzTUQHNScT5Wye27MjKqkZokjUvruW+h9UURE6A3BillIA2WbM5SAalXBaU63pa0UYCynBCJ+a5UVcBxW2hgNeti77XJDKBV8Mdg7FJ80xNMGAlh2S8DVkwaotR8gnG5u0Xmlg3Z8xaFhmAZZQy5vaFiZgIJXZzGaM0dihJbS2GhUga2BHPxsFzawSwrR1iwOYCR0946ryKqiryP9myHmAbSs8yAjov063YFquwAAWX9V6a4Kdcs45t7j13tq3VgdGpa0KtCzq51d7m20+K8+xU29hTIgfWHcAwEW7DmH5TFnQZovgZuc1IK16HXhZfQ7lqK+WctQJtUE8RhVJHSPUTMBO8jFFwGUUC3q6jcZaPJdAOASyoBsKgPJHhYr7q+MKkAly4b949ApM2LqBSXh8LdAI6bXWzy9fI7NkL9hhYJXxlpFwpXQk+IiS1Ghdc8n1ZUU1jRrRtHtYO5+WyaxU6Vm6yI+0g4hAIB3Udh0kcgHKzQ6V5MEFVOQ3WdiFNzBMMSn6JibEuICREnAuSSk8FbFIQZCl2SpX+u9DsETIZHwRhS2AF+eYCJYBUBpdMwtv32lrHDu98ncG7wnEhfMBbtEsMDfm0QkrYi9vhITq9Zg0xNv9FmwIs6z3XtQlgMziA9p4gJyTvEbMvRnyVIpxaoNSbVDJ3YjKNEOv4jhbvKwQ37Eld3up+ZaqjxGyM4CerUdUv3uy-l5gYdmBWxNhgIb93I7Wz6+aL7j3rYoSNkN9tqKq2N12A21ctnJUXjlXSW6CWR5FfJxe26HqcB6GwGcp7QxROlfSDl3APXgwg+65mSb330CmC5-e8VG3FwS9l-o27citnXXTiloH33oUw+m+TmaD3beLPqyYAAdhKHTYVMC88wBOCrxf24sOXKTQnz80f21x628VYhUhyozbvMLZN0Sa5Y6SgI9ioDfDgLyIUAAVJgXOskxdDdYQKEAIgQBaivGgMgKAF1OXG6GzKToASgKwLGBALwPaD-iAOsAuGAEnDAPkK+GAH7MgIMPwOXiAI5gAOp2AkjrS2D1gIADR8C5gwDxDIDgBYoIHTCLA8BlzdDnDIAShNiLC-7FhRCqjIGoH5ChoQyugcFcF4G8H5B+zNgACK-AEA8AYhcCv+GC3AdBmw6mzQCB2YVGBBcBpIyAlEFQIA2YKocABBC47AdBKuzB9CCBAAjvIfAIPLiHQSKOSnAB4h4ggQiI4XEHAKwe0OwUgJwcoSAIsFQtgUMHgQNB0LIU4XAEodwSAI2OsHofaAYUgNnL-oMLdF9O0KXoIMESANXAgYlHAOTusJAaEckbGLgcLI6LAPkGejAIQAmA2PEM2EkfhOYEAA

**Comment by MaathavanJkr** [comment]
Hidden Topics can be seen in bottom now. and further more developments are done. when they become up they get sticky!! i think its perfectly alright check it. i will send a screen shot too

**Comment by MaathavanJkr** [comment]
This time i didnt change the old style of code too, i inserted my code alone!!.

File: source-browser.html [hunk 1]
```diff
@@ -62,7 +78,38 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
     <a class="fixed right-0 bottom-0 mr2 mb2 text-decoration-none h2" id="refresh" href="#">🔄</a>
 
     <script type="text/javascript" src="https://unpkg.com/zepto@1.2.0/dist/zepto.js" integrity="sha384-Cp3V2nlfJJ5aA0ctd1PkfNAEMkXM0EGa6RrmEgiv8D6TCaeZJfUFs/PYfR+B5F5H" crossorigin="anonymous"></script>
+      
     <script type="text/javascript">
+    setInterval(function(){ 
+        $( ".stopic" ).each(function(index) {
```
Comment (wei2912) [inline source-browser.html]: Please adhere to the previous style and remove the spaces within the brackets i.e. change `$( ".stopic" )` to `$(".stopic")`.

Comment (wei2912) [reply source-browser.html]: also what does `stopic` mean?

**Review by wei2912 — CHANGES_REQUESTED** [review]

File: source-browser.html (hunk 1 above)
Comment (MaathavanJkr) [reply source-browser.html]: sticky Topics
shortly stopics

**Comment by wei2912** [comment]
My brief testing showed this:

2019-12-06-222230_scrot

The bottom sticky headings are positioned very low, and I think it'd be better to have them hover such that they touch the bottom of the "frame" (marked out by the scrollbar). This way it becomes more clear that the headings are part of the content. I also suggest adding a translucent black background to the row of the headings so that we are able to see text scrolling behind the headings.

Otherwise, pretty good work so far - just take care of your code style!

EDIT: Did some further testing, there's a point where the bottom headings start to flash.

File: source-browser.html [hunk 2]
```diff
@@ -62,6 +87,44 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
     <a class="fixed right-0 bottom-0 mr2 mb2 text-decoration-none h2" id="refresh" href="#">🔄</a>
 
     <script type="text/javascript" src="https://unpkg.com/zepto@1.2.0/dist/zepto.js" integrity="sha384-Cp3V2nlfJJ5aA0ctd1PkfNAEMkXM0EGa6RrmEgiv8D6TCaeZJfUFs/PYfR+B5F5H" crossorigin="anonymous"></script>
+    <script type="text/javascript">
+      setInterval(function(){
```
Comment (sushain97) [inline source-browser.html]: Please run this code through Prettier. There are many issues.

File: source-browser.html [hunk 3]
```diff
@@ -62,6 +87,44 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
...
+      setInterval(function(){ 
+          $('#content').height($(window).height() - $('#navbar').height() - $('.stickTopic').height() - 40);
+          $('.repos').each(function(index) {
+              var h = $(window).height() - $('.stickTopic').height() - 8;
+              if ($(this).position().top >= h) {
+                  $(this).prev().addClass('stickyb');
+              }
+              else if ($(this).position().top <= h) {  
+                  $(this).prev().removeClass('stickyb');
+                  $(this).prev().css('width', '100%')
```
Comment (sushain97) [inline source-browser.html]: This method .can be chained with the above.

File: source-browser.html [hunk 4]
```diff
@@ -62,6 +87,44 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
     <a class="fixed right-0 bottom-0 mr2 mb2 text-decoration-none h2" id="refresh" href="#">🔄</a>
 
     <script type="text/javascript" src="https://unpkg.com/zepto@1.2.0/dist/zepto.js" integrity="sha384-Cp3V2nlfJJ5aA0ctd1PkfNAEMkXM0EGa6RrmEgiv8D6TCaeZJfUFs/PYfR+B5F5H" crossorigin="anonymous"></script>
+    <script type="text/javascript">
+      setInterval(function(){ 
+          $('#content').height($(window).height() - $('#navbar').height() - $('.stickTopic').height() - 40);
+          $('.repos').each(function(index) {
+              var h = $(window).height() - $('.stickTopic').height() - 8;
```
Comment (sushain97) [inline source-browser.html]: Don't use `var` Prettier will also fix this.

File: source-browser.html [hunk 5]
```diff
@@ -72,16 +135,13 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
           name: topic,
           repoUrl: `https://github.com/${ORGANIZATION}/apertium-${topic}`,
         }));
-
```
Comment (sushain97) [inline source-browser.html]: Please undo the deletion of all these new lines.

File: source-browser.html [hunk 6]
```diff
@@ -92,56 +152,45 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
...
         const normalizedSearchText = $('#search')
           .val()
           .toLowerCase();
-
         $('.topic').remove();
-
         TOPICS.forEach(({ topic, name, repoUrl }, i) => {
-          const repoList = $('<ul class="repos list-reset mt0">');
-
+          const repoList = $('<ul id="'+ name +'" class="repos list-reset mt0">');
```
Comment (sushain97) [inline source-browser.html]: Please use jQuery's built-in attribute object parameter.

**Review by sushain97 — CHANGES_REQUESTED** [review]

File: source-browser.html (hunk 6 above)
Comment (MaathavanJkr) [reply source-browser.html]: Okay!

File: source-browser.html (hunk 4 above)
Comment (MaathavanJkr) [reply source-browser.html]: it didn't!!

File: source-browser.html (hunk 2 above)
Comment (MaathavanJkr) [reply source-browser.html]: Done!

File: source-browser.html (hunk 3 above)
Comment (MaathavanJkr) [reply source-browser.html]: Done! :)

File: source-browser.html (hunk 4 above)
Comment (MaathavanJkr) [reply source-browser.html]: Prettier didnt fixed it, but i do, so done! :)

File: source-browser.html (hunk 5 above)
Comment (MaathavanJkr) [reply source-browser.html]: This time only one change had made in the old script! so done :)

File: source-browser.html (hunk 6 above)
Comment (MaathavanJkr) [reply source-browser.html]: That change isn't necessary.. So Done ;)

**Comment by MaathavanJkr** [comment]
See Now, I just fixed everything i think! :)
https://maathavanjkr.github.io/source-browser.html

**Pull request merged**